R/seroprevalence_data.R

```diff
@@ -120,4 +120,203 @@ prepare_bin_data <- function(serodata) {
+#' @param seed The seed for random number generation.
+#' @return A simulated list of counts following a binomial distribution in accordance with a given 
+#' force of infection and age class sizes.
+#' @examples
+#'\dontrun{
+#'
+#' }
+#' @export
+get_sim_counts <- function(sim_data, foi, size_age_class, seed = 1234) {
+  sim_probabilities <- get_sim_prob(sim_data = sim_data, foi = foi, seed = 1234)
+  sim_counts <- purrr::map_int(sim_probabilities, ~rbinom(1, size_age_class, .))
+
+  return(sim_counts)
+}
+
+#' Function that generates simulated positive counts assuming a known historical force of infection
+#'
+#' @param foi Numeric atomic vector corresponding to the desired Force-of-Infection
+#' @param size_age_class Size of each age group specified by either an atomic
+#' vector of the same size as \code{foi} or an integer.
+#' This corresponds to the number of trials \code{size} in \link[stats]{rbinom}.
+#' @return Dataframe object containing the simulated data generated from \code{foi}
+#' @examples
+#'\dontrun{
+#' size_age_class = 5
+#' foi <- rep(0.02, 50)
+#' sim_data <- generate_sim_data(foi = foi,
+#'                               size_age_class = size_age_class,
+#'                               tsur = 2050,
+#'                               birth_year_min = 2000,
+#'                               survey_label = 'sim_constant_foi')
+#' }
+#' @export
+generate_sim_data <- function(foi,
+                              size_age_class,
+                              tsur,
+                              birth_year_min,
+                              survey_label,
+                              test = "fake",
+                              antibody = "IgG",
+                              seed = 1234
+                              ){
+    sim_data <- data.frame(birth_year = c(birth_year_min:(tsur - 1))) %>%
+        mutate(tsur = tsur,
+            country = 'None',
+            test = test,
+            antibody = antibody,
+            survey = survey_label,
+            age_mean_f = tsur - birth_year)
+    sim_data <- sim_data %>%
+        mutate(counts = get_sim_counts(sim_data, foi, size_age_class, seed = seed),
+            total = size_age_class) %>%
+        prepare_serodata(add_age_mean_f = FALSE)
+
+    return(sim_data)
+}
+
+#' Method for constructing age-group variable from age column
+#'
+#' This function was taken from \link[vaccineff]{get_age_group}.
+#' This method splits an age interval from min_val to max_val into
+#' (max_val-min_val)/step intervals.
+#' By default min_val is set 0, however it can be assigned by
+#' convenience.
+#' If the method finds ages greater or equal than max_val
+#' it assigns the string ">{max_val}".
+#' To avoid errors it is necessary to set step < max_val.
+#' It is also suggested to choose the step such
+#' that max_val%%(step+1) = 0.
+#' @param data dataset with at least a column containing the age
+#' information
+#' @param col_age name of the column containing the age
```

Comment:
I don't understand why this function takes a dataset (dataframe?) as input. Wouldn't supplying an age vector make more sense (since it uses no other info in the dataframe)?

Reply:
Addressed by commit [c2c8661](https://github.com/epiverse-trace/serofoi/commit/c2c8661606f05a4e979c2615514a63f4fa0572c1). Now the function takes an age vector as input.